bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -107,12 +107,13 @@ public void updated(Provider<E> provider, E oldElement, E element) {
     }
 
     @Override
-    public void add(E element) {
+    public E add(E element) {
         if (this.managedProvider != null) {
             this.managedProvider.add(element);
         } else {
             throw new IllegalStateException("ManagedProvider is not available");
         }
+        return element;
```

Comment:
I would prefer to move this line in the if branch, so it is more clear, "return element" is done if a managed provider is available only